apps/settings/js/panels/usb_storage/usb_transfer.js

```diff
@@ -71,7 +169,8 @@ define(function(require) {
       } else if (mode === this.MODE_MTP &&
         protocol === this.PROTOCOL_UMS) {
         if (this._partialUmsSupport) {
-          return DialogService.show('partial-ums-warning').then((result) => {
```

Comment:
This change is fine but _changeMode itself seems to resolve synchronously though the operation that sets mode is async. I'm not super partial though, so your call if you think this needs/doesn't need to be made less confusing.